linked_list/.~c9_invoke_17vQM9.py

```diff
@@ -0,0 +1,172 @@
+class AbstractLinkedList(object):
+    """
+    Abstract class representing the LinkedList inteface you must respect.
+    
+    You must not implement any of the method in this class, and this class
+    must never be instantiated. It's just a "guide" of which methods
+    the LinkedList class should respect.
+    """
+
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __len__(self):
+        raise NotImplementedError()
+
+    def __iter__(self):
+        raise NotImplementedError()
+
+    def __getitem__(self, index):
+        raise NotImplementedError()
+
+    def __add__(self, other):
+        raise NotImplementedError()
+
+    def __iadd__(self, other):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        raise NotImplementedError()
+
+    def append(self, element):
+        raise NotImplementedError()
+
+    def count(self):
+        raise NotImplementedError()
+
+    def pop(self, index=None):
+        raise NotImplementedError()
+
+
+class Node(object):
+    """
+    Node class representing each of the linked nodes in the list.
+    """
+
+    def __init__(self, elem, next=None):
+        self.elem = elem
+        self.next = next
+
+    def __str__(self):
+        return str(self.elem)
+
+    def __eq__(self, other):
+        if isinstance(other, LinkedList):
+            return self.elem == other.elem #compare node.elem to value passed to it
+        else:
+            return self.elem == other
+
+    def __repr__(self):
+        return "<Node {}>".format(self.elem)
+
+
+class LinkedList(AbstractLinkedList):
+    """
+    Implementation of an AbstractLinkedList inteface.
+    """
+
+    def __init__(self, elements=None):
+        # start at none
+        self.start = None
+        self.end = None
+        #[1,2], self.start (Node(1)) >> Node1.next now pts to Node(2) 
+        # set elements to empty list
+        elements = [] if not elements else elements
+        for val in elements:
+            self.append(val)
```

Comment:
I like this. We initialized our linked list entirely within the \_\_init\_\_ method without making any calls to append(). This is cleaner.